core/src/main/java/io/opencensus/tags/TagContext.java

```diff
@@ -47,30 +49,39 @@
   }
 
   /**
-   * Returns a builder based on this {@code TagContext}.
+   * Returns the current {@code TagContext}.
    *
-   * @return a builder based on this {@code TagContext}.
+   * @return the current {@code TagContext}.
    */
-  public Builder toBuilder() {
-    return new Builder(getTags());
+  public static TagContext getCurrentTags() {
```

Comment:
I'm not sure.  On one hand, adding the indirection would give us more flexibility to change TagContext in the future, for example, by adding error reporting or validation of tag metadata.  On the other hand, it would be nice to keep the API simple by making TagContext a final class that users don't need to mock.

I would rather use the simpler design for now and add a factory after we decide we need it.